registrations/forms.py

```diff
@@ -95,4 +99,28 @@ def clean_clinic_code(self):
         code = self.cleaned_data["clinic_code"]
         if not code.isdigit():
             raise forms.ValidationError(self.CLINIC_CODE_ERROR_MESSAGE)
+
+        #  Check clinic code exists
+        try:
+            response = requests.get(
+                "%sNCfacilityCheck" % settings.JEMBI_URL,
+                params={"criteria": "value:%s" % code},
+                auth=settings.JEMBI_AUTH,
+            )
+            response.raise_for_status()
+        except requests.exceptions.HTTPError:
+            errors = self.request.session.get("jembi_api_errors", 0)
+            self.request.session["jembi_api_errors"] = errors + 1
+            if errors + 1 >= 3:
+                logging.exception("Jembi API error limit reached")
+            raise forms.ValidationError(
+                "There was an error checking your details. Please try again."
+            )
+
+        data = response.json()
```

Comment:
I'm thinking that we should probably treat them sending us a non-json response the same as any other error.

So moving this line to the try/except block, and adding `JSONDecodeError` to the list of caught errors.